### PR TITLE
Weather plugin fixes

### PR DESCRIPTION
**Pull request by steven676**
Fixes a couple of issues with display in the weather plugin.

### CODE REVIEW

File: ircbot/plugin/weather.py [hunk 1]
```diff
@@ -58,11 +57,14 @@ def deg_to_compass(deg):
 
 
 def icon(temp, unit='f'):
```
Comment (kpengboy) [inline ircbot/plugin/weather.py]: Nit: do we ever call this function (or `color`) without a `unit` argument? If not, then there is not much point in making it optional.

File: ircbot/plugin/weather.py [hunk 2]
```diff
@@ -146,7 +151,11 @@ def get_summary(api_key, location, unit='f'):
         winddeg = j['wind']['deg']
         winddeg_text = f' {winddeg}° ({deg_to_compass(winddeg)})'
 
-    wind_text = f'{windspeed}m/s{winddeg_text}'
+    windunit_text = 'm/s'
+    if unit == 'f':
+        windunit_text = 'mph'
+
+    wind_text = f'{windspeed}{windunit_text}{winddeg_text}'
```
Comment (kpengboy) [inline ircbot/plugin/weather.py]: Nit: would put space between number and unit

File: ircbot/plugin/weather.py [hunk 3]
```diff
@@ -73,14 +75,17 @@ def bold(text):
 
 
 def color(temp, text=None, unit='f'):
+    if unit == 'c':
+        temp_f = c2f(temp)
+    else:
+        temp_f = temp
```
Comment (kpengboy) [inline ircbot/plugin/weather.py]: Nit: may be more compact using a ternary condition?

`temp_f = c2f(temp) if unit == 'c' else temp`

File: ircbot/plugin/weather.py [hunk 4]
```diff
@@ -20,9 +20,8 @@ def weather(bot, msg):
         msg.respond(f'idk where {where} is')
 
 
-def f2c(temp):
-    return int((temp - 32) * 5 / 9)
-
+def c2f(temp):
+    return temp * 9/5 + 32
```
Comment (kpengboy) [inline ircbot/plugin/weather.py]: Make sure to insert 1 more blank line so this file passes style check (and therefore doesn't break the build)

File: ircbot/plugin/weather.py [hunk 5]
```diff
@@ -58,11 +57,14 @@ def deg_to_compass(deg):
 
 
 def icon(temp, unit='f'):
+    if unit == 'c':
+        temp = c2f(temp)
+
     # it doesn't get cold in the bay area, anything less than 60°F is a snowman
-    if temp < (60 if unit == 'f' else f2c(60)):
+    if temp < 60:
```
Comment (Aesthetic) [inline ircbot/plugin/weather.py]: does this still check correctly? it looks to be F only

File: ircbot/plugin/weather.py (hunk 1 above)
Comment (steven676) [reply ircbot/plugin/weather.py]: Nope -- will make the argument mandatory, as suggested.

File: ircbot/plugin/weather.py (hunk 2 above)
Comment (steven676) [reply ircbot/plugin/weather.py]: I was following the original behavior here -- we can certainly have a space, though.

File: ircbot/plugin/weather.py (hunk 5 above)
Comment (steven676) [reply ircbot/plugin/weather.py]: There's a c2f() conversion for temp just above.

**Comment by steven676** [comment]
Okay, now with the requested changes (plus another style fix to shut the checks up) ...

File: ircbot/plugin/weather.py (hunk 5 above)
Comment (Aesthetic) [reply ircbot/plugin/weather.py]: that's the issue, it still checks against the hardcoded 60 F value

File: ircbot/plugin/weather.py [hunk 6]
```diff
@@ -72,15 +75,14 @@ def bold(text):
     return f'\x02{text}\x0F'
 
 
-def color(temp, text=None, unit='f'):
-    if text is None:
-        text = '{}°{}'.format(temp, unit.capitalize())
+def color(temp, unit):
+    temp_f = c2f(temp) if unit == 'c' else temp
+
+    text = '{}°{}'.format(temp, unit.capitalize())
```
Comment (ethanhs) [inline ircbot/plugin/weather.py]: Optional nit: this can be changed into an f string! Something like 

`text = f'{temp}°{unit.capitalize()}'`

Comment (steven676) [reply ircbot/plugin/weather.py]: That's a pretty big win for readability, let's do that.

**Pull request merged**